earthfile2llb/withdockerrun.go

```diff
@@ -335,7 +335,8 @@ func (wdr *withDockerRun) solveImage(ctx context.Context, mts *states.MultiTarge
 			llb.Platform(llbutil.DefaultPlatform()),
 			llb.WithCustomNamef("[internal] docker tar context %s %s", opName, sessionID),
 		)
-		wdr.c.mts.Final.LocalDirs[string(solveID)] = outDir
+		// Add directly to build context so that if a later statement forces execution, the images are avaliable.
+		wdr.c.opt.BuildContextProvider.AddDir(string(solveID), outDir)
```

Comment:
Note that this isn't the only place where we add things to `LocalDirs` in the converter. `FromDockerfile` also does that and would be prone to the same kind of issue. We should try to remove `SingleState.LocalDirs` altogether and instead use `BuildContextProvider.AddDir` directly everywhere.

Reply:
FromDockerfile never adds things, it only copies what exists. Because its not adding anything new, it shouldnt be affected?